src/Workspaces/Core/Portable/Shared/Extensions/ISymbolExtensions.cs

```diff
@@ -865,6 +865,12 @@ public static ITypeSymbol GetSymbolType(this ISymbol symbol)
                 return aliasSymbol.Target as ITypeSymbol;
             }
 
+            var methodSymbol = symbol as IMethodSymbol;
```

Comment:
This doesn't seem right.  IIRC this didn't return .ReturnType because it's equally valid to view the type of an IMethodSymbol as being the delegate type with the signature that matches the method.  i.e. if i have:

```NewMethod(someMethod)```

we'd want the type of the first parameter to be the Func/Action type that matches 'someMethod'.  As we're not invoking someMethod we don't want it to be the return type.